### PR TITLE
infer dimension from geometry

### LINKED ISSUE #41: astra utils to infer geometry.dimension from geometry
The conversion from `ImageGeometry` and `AcquisitionGeometry` to astra geometry uses the parameter `dimension` of the `ImageGeometry` and `AcquisitionGeometry` instances. However, it should better infer if it's `2D` or `3D` (+channel) by reading the rest of the geometry members.

Also, the member `dimension` has been used loosely and without any checks. Currently it could contain anything and it would only break the conversion of the geometry [here](https://github.com/vais-ral/CCPi-astra/blob/master/Wrappers/Python/ccpi/astra/utils/convert_geometry_to_astra.py#L6) and [here](https://github.com/vais-ral/CCPi-astra/blob/master/Wrappers/Python/ccpi/astra/utils/convert_geometry_to_astra.py#L29).

See here for further considerations: https://github.com/vais-ral/CCPi-Framework/issues/408

### PR DESCRIPTION
**Pull request by paskino**
closes #41

```python
from ccpi.astra.utils import *
from ccpi.framework import *
import numpy as np

n_angles = 10
angles = np.linspace(0, np.pi, n_angles, dtype=np.float32)

ag = AcquisitionGeometry('cone', 'edo', pixel_num_h=20, pixel_num_v=2, angles=angles, 
                         dist_source_center = 312.2, 
                         dist_center_detector = 123. )
#ag = AcquisitionGeometry('parallel', 'edo', pixel_num_h=20, pixel_num_v=1, angles=angles )
ig = ImageGeometry(20,20)
astra_ig, astra_ag = convert_geometry_to_astra(ig, ag)
print (astra_ig)
print (astra_ag)
```
seems to work right.

### CODE REVIEW

**Comment by epapoutsellis** [comment]
running [PDHG_TV_Tomo](https://github.com/vais-ral/CIL-Demos/blob/master/Tomography/Simulated/Single%20Channel/PDHG_TV_Tomo2D.py)

  File "/anaconda3/envs/ccpi_vaggelis/lib/python3.6/site-packages/ccpi/astra/utils/convert_geometry_to_astra.py", line 20, in convert_geometry_to_astra
    raise ValueError('Number of pixels at detector on the vertical axis must be > 1. Got {}'.format(vert))

ValueError: Number of pixels at detector on the vertical axis must be > 1. Got 0

File: Wrappers/Python/ccpi/astra/utils/convert_geometry_to_astra.py [hunk 1]
```diff
@@ -2,8 +2,24 @@
...
+       :param sinogram_geometry: ccpi.framework.AcquisitionGeometry
+       
+       :returns ASTRA volume and sinogram geometry'''
+
+    # determine if the geometry is 2D or 3D
+    horiz = sinogram_geometry.pixel_num_h
+    vert  = sinogram_geometry.pixel_num_v
+    if vert > 1:
+        dimension = '3D'
+    elif vert == 1:
```
Comment (epapoutsellis) [inline Wrappers/Python/ccpi/astra/utils/convert_geometry_to_astra.py]: or vert==0

**Comment by jakobsj** [comment]
I agree we can possibly simplify and get rid of 2D/3D, however let's not
rush this and risk breaking stuff. We also have distinct 2D and 3D astra
operators that we want to consider. It's not clear that a 3d case with a
single vertical slice is the same as 2d necessarily. Further we may at some
point consider reconstruction of 3d volumes from line detector data. So we
should think a bit more about this.

**Comment by paskino** [comment]
I believe that we need to fix this for the current release and then think about it in the long term. 

The parameter `dimension` is not used correctly, it's never checked. The only place it's used/checked is here, so I strongly suggest we get this repo to deal with it. 

Agree on @epapoutsellis correction as 0 is the [default size](https://github.com/vais-ral/CCPi-Framework/blob/master/Wrappers/Python/ccpi/framework/framework.py#L224).

**Comment by epapoutsellis** [comment]
This is only related to the convert to astra geometry, so we have the following cases

1. 2D ( no channels): Consider as 2D
1. 2D + Channels : Still we consider it as 2D
1. 3D (no channels): Consider as 3D
1. 3D + channels:  Still we consider it as 3D

If we have 

1. 3D and no channels , the subset(vertical) will produce 2D, that is ok
1. 3D and channels, the subset(vertical) will produce 2D + channels, **not 3D**

We can leave as it is (have the dimension) and do the change of @paskino , otherwise it will break the [FBP](https://github.com/vais-ral/CCPi-astra/blob/master/Wrappers/Python/ccpi/astra/processors/FBP.py#L36)

**Comment by paskino** [comment]
I amended the PR. Please check if it fixes the problem.

**Comment by paskino** [comment]
> OK then, with vert 0. I think actually at present we may risk conflicting geometry information, say by using dimension 2D, but giving x, y, z numbers of pixels. (Perhaps that is what you are referring to with dimension not being checked properly.) By removing dimension, I think we get rid of that redundancy.

That's exactly it. The parameter is manually set, and can be set to anything. Then the code in `convert_geometry_to_astra` relies on it. That's why I'm pushing to not use it.

**Pull request merged**